File: Data Structures & Algorithms/pow-x-n/submission-1.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        ans = 1
        N = abs(n)
        while N > 0:
            if N & 1:
                ans *= x
            x *= x
            N >>= 1
        return ans if n >= 0 else 1 / ans

File: Data Structures & Algorithms/string-encode-and-decode/submission-1.java
class Solution {

    public String encode(List<String> strs) {
        StringBuilder sb = new StringBuilder();
        for (String st : strs) {
            sb.append(st.length()).append("#").append(st);
        }
        return sb.toString();
    }

    public List<String> decode(String str) {
        List<String> ans = new ArrayList<>();
        int i = 0;
        while (i < str.length()) {
            int j = i;
            while (str.charAt(j) != '#') {
                j++;
            }
            int len = Integer.parseInt(str.substring(i, j));
            i = j + 1;
            j = i + len;
            ans.add(str.substring(i, j));
            i = j;
        }
        return ans;
    }
}
